backend/app/health_handler.go

```diff
@@ -0,0 +1,124 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"kubecloud/models/sqlite"
+
+	"github.com/gin-gonic/gin"
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	proxyTypes "github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/pkg/types"
+)
+
+type HealthStatus struct {
+	Status  string `json:"status"`
+	Message string `json:"message,omitempty"`
+}
+
+type HealthChecker func(ctx context.Context) HealthStatus
+
+func statusFromError(err error) HealthStatus {
+	if err != nil {
+		return HealthStatus{Status: "unhealthy", Message: err.Error()}
+	}
+	return HealthStatus{Status: "healthy"}
+}
+
+func simpleHealthCheck(client interface{}, checkFn func() error) HealthStatus {
+	if client == nil {
+		return HealthStatus{Status: "unhealthy", Message: "client not initialized"}
+	}
+	if err := checkFn(); err != nil {
+		return HealthStatus{Status: "unhealthy", Message: err.Error()}
+	}
+	return HealthStatus{Status: "healthy"}
+}
+
+func (h *Handler) checkDatabase(ctx context.Context) HealthStatus {
+	sqliteDB, ok := h.db.(*sqlite.Sqlite)
+	if !ok {
+		return HealthStatus{Status: "unhealthy", Message: "not a sqlite DB"}
+	}
+	db, err := sqliteDB.SQLDB()
+	if err != nil {
+		return statusFromError(err)
+	}
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
```

Comment:
The timeout is reasonable, however, please open a ticket to have all of these durations configured from config.